src/NexusMods.CrossPlatform/Process/ProcessFactory.cs

```diff
@@ -9,18 +12,93 @@ namespace NexusMods.CrossPlatform.Process;
 public class ProcessFactory : IProcessFactory
 {
     private readonly ILogger _logger;
+    private readonly IFileSystem _fileSystem;
+    private readonly AbsolutePath _processLogsFolder;
 
     /// <summary>
     /// Constructor.
     /// </summary>
-    public ProcessFactory(ILogger<ProcessFactory> logger)
+    public ProcessFactory(
+        ILogger<ProcessFactory> logger,
+        IFileSystem fileSystem,
+        ISettingsManager settingsManager)
     {
         _logger = logger;
+        _fileSystem = fileSystem;
+
+        _processLogsFolder = LoggingSettings.GetLogBaseFolder(fileSystem.OS, fileSystem).Combine("ProcessLogs");
+        _logger.LogInformation("Using process log folder at {Path}", _processLogsFolder);
+
+        _processLogsFolder.CreateDirectory();
+
+        var loggingSettings = settingsManager.Get<LoggingSettings>();
+        var retentionSpan = loggingSettings.ProcessLogRetentionSpan;
+
+        var filesToDelete = _processLogsFolder
+            .EnumerateFiles()
+            .Where(x => DateTime.Now - x.FileInfo.CreationTime >= retentionSpan)
+            .ToArray();
+
+        foreach (var file in filesToDelete)
+        {
+            try
+            {
+                file.Delete();
+            }
+            catch (Exception e)
+            {
+                _logger.LogError(e, "Failed to delete expired log file {Path}", file);
+            }
+        }
```

Comment:
What to refactor this into a private method for clarity since it's not strictly related?